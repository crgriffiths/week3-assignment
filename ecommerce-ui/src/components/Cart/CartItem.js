import React from 'react';

class CartItem extends React.Component {
  constructor(props) {
    super(props)
  }
  render() {
    return (
      <li>
        <div className="d-flex align-items-center">
          <div className="col-3 px-1">
            <img src={this.props.image}/>
          </div>
          <p className="col-6 px-1">{this.props.title}</p>
          <p className="col-2 px-1">${this.props.cost}</p>
          <button className="remove-listing" onClick={()=> this.props.onRemoveFromCart(this.props.index)}></button>
        </div>
      </li>
    )
  }
}

export default CartItem;