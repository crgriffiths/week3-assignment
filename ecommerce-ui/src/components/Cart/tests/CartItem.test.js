import React from 'react';
import CartItem from '../CartItem';
import renderer from 'react-test-renderer';

test('Cart item renders in cart when added', () => {
  const component = renderer.create (
    <CartItem 
      key={0}
      index={0}
      image='https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&amp;ixid=eyJhcHBfaWQiOjEyMDd9&amp;s=2ca889535ea01f912f94ac4ddf0034e0&amp;auto=format&amp;fit=crop&amp;w=500&amp;q=80'
      title='Centrally-located Manhattan studio'
      cost={105}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});