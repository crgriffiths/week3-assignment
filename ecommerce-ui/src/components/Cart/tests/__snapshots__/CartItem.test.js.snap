// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart item renders in cart when added 1`] = `
<li>
  <div
    className="d-flex align-items-center"
  >
    <div
      className="col-3 px-1"
    >
      <img
        src="https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80"
      />
    </div>
    <p
      className="col-6 px-1"
    >
      Centrally-located Manhattan studio
    </p>
    <p
      className="col-2 px-1"
    >
      $
      105
    </p>
    <button
      className="remove-listing"
      onClick={[Function]}
    />
  </div>
</li>
`;
